spec/services/migration/migrators/application_spec.rb

```diff
@@ -0,0 +1,71 @@
+require "rails_helper"
+
+RSpec.describe Migration::Migrators::Application do
+  let(:instance) { described_class.new }
+
+  subject { instance.call }
+
+  describe "#call" do
+    let(:ecf_npq_application1) { create(:ecf_migration_npq_application) }
+    let(:ecf_npq_application2) { create(:ecf_migration_npq_application) }
+
+    before do
+      create(:application, ecf_id: ecf_npq_application1.id)
+      create(:application, ecf_id: ecf_npq_application2.id)
+
+      create(:data_migration, model: :application)
+    end
+
+    context "when all attributes values of interest matches" do
+      before do
+        allow_any_instance_of(Migration::Ecf::NpqApplication).to receive(:attributes).and_return({})
```

Comment:
This feels a bit heavy handed on the mocking as it will skip all the comparison logic in the migrator; would it be better to set the application attributes explicitly for the state where they should match?